### PR TITLE
Cleanup of package structure, descriptor files, and readmes.

### PR DESCRIPTION
**Pull request by cfernald**

### CODE REVIEW

File: IpmiFeaturePkg/Library/IpmiBaseLibMm/IpmiBaseLibMm.inf [hunk 1]
```diff
@@ -7,14 +7,14 @@
 
 [Defines]
   INF_VERSION                    = 0x00010005
-  BASE_NAME                      = SmmIpmiBaseLib
+  BASE_NAME                      = IpmiBaseLibMm
```
Comment (spbrogan) [inline IpmiFeaturePkg/Library/IpmiBaseLibMm/IpmiBaseLibMm.inf:10]: not sure about this one.  You are only supporting SMM..not MM.

**Comment by spbrogan** [comment]
Not sure if this is the PR to solve this and/or if this issue came from upstream base code or was introduced in your changes but there is a lot of mixtures of ipmi, bmc, and ipmibmc.   Example: PeiGetIpmiBmcStatus, PeiIpmiSendCommand, SendDataToBmcPort, InitializeIpmiTransportHardware

**Comment by cfernald** [comment]
> 


> Not sure if this is the PR to solve this and/or if this issue came from upstream base code or was introduced in your changes but there is a lot of mixtures of ipmi, bmc, and ipmibmc. Example: PeiGetIpmiBmcStatus, PeiIpmiSendCommand, SendDataToBmcPort, InitializeIpmiTransportHardware

Yeah, this inconsistency is in the upstream code. Because IPMI is only used to talk to the BMC, it's a somewhat meaningless naming. I think the only time it's appropriate is in the context of a module specific to BMC that's not strictly IPMI such as the SMBus library used by SSIF. I will go ahead and clean this up in this PR.

**Review by spbrogan — APPROVED** [review]
Once changes are made this looks like good progress

**Pull request merged**